src/main/java/templates/mapPages.soy

```diff
@@ -50,12 +51,12 @@
 
                         <div class="selection-controls-container">
            
-                            {for $groupName in $groupNames}
+                            {for $groupMap in $groups}
                                 <div>
```

Comment:
Can you refactor this so that this is its own template? It'll make this easier to read.